src/gmt_support.c

```diff
@@ -10485,10 +10485,18 @@ int gmtlib_decorate_specs (struct GMT_CTRL *GMT, char *txt, struct GMT_DECORATE
 					}
 					else if (p[1] == 'k') {	/* Custom symbol - separate file from size */
 						char *s = strrchr (p, '/');
-						strncpy (G->size, &s[1], GMT_LEN64-1);
-						s[0] = '\0';	/* Truncate size */
-						strncpy (G->symbol_code, &p[1], GMT_LEN64-1);
-						s[0] = '/';	/* Restore size */
+						if (s == NULL) {	/* No slash, no size */
+							GMT_Report (GMT->parent, GMT_MSG_ERROR, "Option -S~: Custom symbol %s not given any size\n", &p[2]);
+							bad++;
+						}
+						else {
+							strncpy (G->size, &s[1], GMT_LEN64-1);
```

Comment:
If a unix full file name was passed and no size, this will pick the file name as if it was the size, no?

Reply:
Yes, looks like it.  I need to add a check if that string is numeric (there is a function for this).

Reply:
Check added.